### DISCUSSION TITLE
JBang and annotation processors?

### DISCUSSION THREAD

**Discussion by Klaboe** [jbangdev/jbang#1529] [Q&A] [answered]
So this is more a "try to understand the technology" rather than "Should jbang support this" kind of query

So i know there is quarkus support but as i understand it, this is maintaned by the quarkus team and is not a thing JBang handles?

I started working with Micronaut a while back, and thought i would give it a go with JBang; the dependencies sync as they should but when i run it, it starts, but the annotated functionallity does not kick in.

The kicker is that if i run it in Eclipse (with the jbang plugin installed) they kick in just fine, and my restcontroller answers as it should when i call the URI. But running it vanilla from the commandline i just get a 404 which suggest the routing does not get hooked up and i would assume this is because lack of annotation processing

So any suggestion how i can get this to work with vanila jbang outside my IDE?

**Comment by maxandersen** [comment, marked as answer]
jbang just let javac pick up annotation processors from dependencies.

so it should just work unless micronaut has different kind of annotation processors that requires manual registration?

**Reply by Klaboe** [reply]
Huh; interesting. So this is what i got:

Main class 
<img width="281" alt="image" src="https://user-images.githubusercontent.com/126390/209205127-9eff47fb-5a46-4238-8de0-e0588d02e6d6.png">

And then my Controller:
<img width="256" alt="image" src="https://user-images.githubusercontent.com/126390/209205240-96224d79-06be-4504-8ba7-6abace46f39a.png">

If i run `jbang run Main.java` standing in jbang.test ; it starts the webserver but i get a 404 for /hello . But you say it should pick up the annotations here? If i just run this in eclipse it gets picked up and i get a 200 response

**Reply by maxandersen** [reply]
If the annotation processor is included in that classpath it should just work. 

Can't test it as not near computer at the moment.

**Reply by maxandersen** [reply]
could not get your example to build/compile so made my own based on it https://gist.github.com/maxandersen/eab8d454ed1583888c81a1820a1394b1

It works as long as you put instructions in Main.java that you want to add additional sources.

In this I added `//SOURCES MicroController.java` but you could also just add `//SOURCES **/*.java` if you want all java files.

**Reply by Klaboe** [reply]
Yes this seems to be exactly what i was after; and also probably what the IDE-plugin does behind the scenes when i run it from there?

Thank you so much for the insight 🙌

**Reply by maxandersen** [reply]
Depends how you opened it in your IdE. But yeah it probably included all .java files it found rather than just the primary one you chose to run with jbang.

**Reply by fbricon** [reply]
Looks like a manifestation of https://github.com/jbangdev/jbang/issues/1502
